src/cmd/linuxkit/run_vmware.go

```diff
@@ -115,54 +123,56 @@ func runVMware(args []string) {
 			log.Fatalf("Unable to find %s within the $PATH", vmrunPath)
 		}
 		vmrunPath = fullVMrunPath
-		vmrunArgs = []string{"-T", "ws", "start", prefix + ".vmx"}
+		vmrunArgs = []string{"-T", "ws", "start"}
 	}
 
 	// Check vmrunPath exists before attempting to execute
 	if _, err := os.Stat(vmrunPath); os.IsNotExist(err) {
 		log.Fatalf("ERROR VMware executables can not be found, ensure software is installed")
 	}
 
-	if *vmwareDisk != "" {
+	if *disk == "" && *diskSz != "" {
+		*disk = filepath.Join(*state, "disk.vmdk")
+	}
+	if *disk != "" {
 		// Check vmDiskManagerPath exist before attempting to execute
 		if _, err := os.Stat(vmDiskManagerPath); os.IsNotExist(err) {
 			log.Fatalf("ERROR VMware Disk Manager executables can not be found, ensure software is installed")
 		}
 
 		// If disk doesn't exist then create one, error if disk is unreadable
-		if _, err := os.Stat(*vmwareDisk); err != nil {
+		if _, err := os.Stat(*disk); err != nil {
 			if os.IsPermission(err) {
-				log.Fatalf("Unable to read file [%s], please check permissions", *vmwareDisk)
+				log.Fatalf("Unable to read file [%s], please check permissions", *disk)
 			}
 			if os.IsNotExist(err) {
-				log.Infof("Creating new VMware disk [%s]", *vmwareDisk)
-				vmDiskCmd := exec.Command(vmDiskManagerPath, "-c", "-s", *vmwareDiskSize, "-a", "lsilogic", "-t", "0", *vmwareDisk)
+				log.Infof("Creating new VMware disk [%s]", *disk)
+				vmDiskCmd := exec.Command(vmDiskManagerPath, "-c", "-s", *diskSz, "-a", "lsilogic", "-t", "0", *disk)
 				if err = vmDiskCmd.Run(); err != nil {
-					log.Fatalf("Error creating disk [%s]:  %s", *vmwareDisk, err.Error())
+					log.Fatalf("Error creating disk [%s]:  %s", *disk, err.Error())
 				}
 			}
 		} else {
-			log.Infof("Using existing disk [%s]", *vmwareDisk)
+			log.Infof("Using existing disk [%s]", *disk)
 		}
 	}
 
 	// Build the contents of the VMWare .vmx file
-	vmx := buildVMX(*vmwareCPUs, *vmwareMem, *vmwareDisk, prefix)
-
+	vmx := buildVMX(*cpus, *mem, *disk, prefix)
 	if vmx == "" {
 		log.Fatalf("VMware .vmx file could not be generated, please confirm inputs")
 	}
 
 	// Create the .vmx file
-	err := ioutil.WriteFile(prefix+".vmx", []byte(vmx), 0644)
-
+	vmxPath := filepath.Join(*state, "linuxkit.vmx")
```

Comment:
It does. I just tested by copying `examples/vmware.yml` to `examples/vmware2.yml` and build and run